modules/pubsub/src/main/java/com/spotify/ffwd/pubsub/PubsubPluginSink.java

```diff
@@ -119,22 +122,32 @@ public void onSuccess(String messageId) {
   public AsyncFuture<Void> sendMetrics(Collection<Metric> metrics) {
     logger.debug("Sending {} metrics", metrics.size());
 
-    try {
-      final ByteString m = ByteString.copyFrom(serializer.serialize(metrics, writeCache));
-
-      if (m.size() > MAX_BATCH_SIZE_BYTES) {
-        logger.info("Above byte limit, resizing batch");
-        int times = (int)Math.ceil(m.size()/MAX_BATCH_SIZE_BYTES);
-        List<List<Metric>> collections = Lists.partition(new ArrayList<>(metrics), metrics.size()/times);
-        for (List<Metric> l: collections) {
-          final ByteString mResize = ByteString.copyFrom(serializer.serialize(l, writeCache));
-          publishPubSub(mResize);
+    if (metrics.size() < MAX_INPUT_METRICS) {
+      try {
+        final ByteString m = ByteString.copyFrom(serializer.serialize(metrics, writeCache));
+        if (m.size() > MAX_BATCH_SIZE_BYTES) {
+          logger.info("Above byte limit {}, size was {}; resizing batch", MAX_BATCH_SIZE_BYTES,
+              m.size());
+          int times = (int) Math.ceil(m.size() / MAX_BATCH_SIZE_BYTES);
+          List<List<Metric>> collections =
+              Lists.partition(new ArrayList<>(metrics), metrics.size() / times);
+          for (List<Metric> l : collections) {
+            try {
+              final ByteString mResize = ByteString.copyFrom(serializer.serialize(l, writeCache));
+              publishPubSub(mResize);
+            } catch (Exception e) {
+              logger.error("Failed to serialize batch of metrics: ", e);
```

Comment:
should we throw an HttpException where we're catching these exceptions?

Reply:
I don't think this is in the path of HTTP requests, the input plugins are done once ffwd has buffered the metrics internally.

Reply:
I don't think I can throw an exception at that point, we'd need to figure out the size sooner somewhere upstream

Reply:
lol hey william